cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_train_hist_oneapi_impl.i

```diff
@@ -76,32 +77,12 @@ static services::String getFPTypeAccuracy()
     return services::String();
 }
 
-inline char * utoa(uint32_t value, char * buffer, size_t buffer_size)
-{
-    size_t i = 0;
-    while (value && i < buffer_size - 1)
-    {
-        size_t rem  = value % 10;
-        buffer[i++] = 48 + rem;
-        value /= 10;
-    }
-
-    for (size_t j = 0; j < i - j - 1; j++)
-    {
-        char tmp          = buffer[j];
-        buffer[j]         = buffer[i - j - 1];
-        buffer[i - j - 1] = tmp;
-    }
-    buffer[i] = 0;
-    return buffer;
-}
-
 static services::String getBuildOptions(size_t nClasses)
 {
-    const uint32_t valBufSize = 16;
-    static char valBuffer[valBufSize];
+    char buffer[DAAL_MAX_STRING_SIZE] = { 0 };
+    const auto written                = daal::services::daal_int_to_string(buffer, DAAL_MAX_STRING_SIZE, static_cast<int32_t>(nClasses));
```

Comment:
Do we need bound checking for 'static_cast<int32_t>(nClasses)' ?

Reply:
thanks, fixed